agents/supervisor/supervisor.go

```diff
@@ -96,9 +99,6 @@ func NewSupervisor(ctx context.Context, paths *config.Paths, ports *config.Ports
 
 // SetState starts or updates all agents placed in args and stops all agents not placed in args, but already run.
 func (s *Supervisor) SetState(state *agentpb.SetStateRequest) {
-	s.m.Lock()
-	defer s.m.Unlock()
```

Comment:
Why we drop that global lock, but replace it with individual locks in each method? It looks like we can loose consistency that way, and also introduce more complexity.